config/yaml.go

```diff
@@ -273,36 +306,34 @@ func (n *yamlNode) Children() []*yamlNode {
 	return nodes
 }
 
-func unmarshalYAMLValue(reader io.ReadCloser, value interface{}, lookUp lookUpFunc) error {
-	raw, err := ioutil.ReadAll(reader)
-	if err != nil {
-		return errors.Wrap(err, "failed to read the yaml config")
+func (n *yamlNode) dfs(expand func(string) string) {
```

Comment:
better to name it something like applyOnAllNodes or expandNodeValues etc. its cool that it uses dfs but does not add much to the readability.